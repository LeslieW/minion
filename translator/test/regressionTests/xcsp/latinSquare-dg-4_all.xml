<?xml version="1.0" encoding="UTF-8"?>

<instance>
<presentation name="?" maxConstraintArity="4" format="XCSP 2.0"/>

<domains nbDomains="1">
<domain name="D0" nbValues="4">0..3</domain>
</domains>

<variables nbVariables="16">
<variable name="V0" domain="D0"/>
<variable name="V1" domain="D0"/>
<variable name="V2" domain="D0"/>
<variable name="V3" domain="D0"/>
<variable name="V4" domain="D0"/>
<variable name="V5" domain="D0"/>
<variable name="V6" domain="D0"/>
<variable name="V7" domain="D0"/>
<variable name="V8" domain="D0"/>
<variable name="V9" domain="D0"/>
<variable name="V10" domain="D0"/>
<variable name="V11" domain="D0"/>
<variable name="V12" domain="D0"/>
<variable name="V13" domain="D0"/>
<variable name="V14" domain="D0"/>
<variable name="V15" domain="D0"/>
</variables>

<constraints nbConstraints="16">
<constraint name="C0" arity="4" scope="V0 V1 V2 V3" reference="global:allDifferent"/>
<constraint name="C1" arity="4" scope="V4 V5 V6 V7" reference="global:allDifferent"/>
<constraint name="C2" arity="4" scope="V8 V9 V10 V11" reference="global:allDifferent"/>
<constraint name="C3" arity="4" scope="V12 V13 V14 V15" reference="global:allDifferent"/>
<constraint name="C4" arity="4" scope="V0 V4 V8 V12" reference="global:allDifferent"/>
<constraint name="C5" arity="4" scope="V1 V5 V9 V13" reference="global:allDifferent"/>
<constraint name="C6" arity="4" scope="V2 V6 V10 V14" reference="global:allDifferent"/>
<constraint name="C7" arity="4" scope="V3 V7 V11 V15" reference="global:allDifferent"/>
<constraint name="C8" arity="4" scope="V12 V1 V6 V11" reference="global:allDifferent"/>
<constraint name="C9" arity="4" scope="V8 V13 V2 V7" reference="global:allDifferent"/>
<constraint name="C10" arity="4" scope="V4 V9 V14 V3" reference="global:allDifferent"/>
<constraint name="C11" arity="4" scope="V0 V5 V10 V15" reference="global:allDifferent"/>
<constraint name="C12" arity="4" scope="V0 V7 V10 V13" reference="global:allDifferent"/>
<constraint name="C13" arity="4" scope="V1 V4 V11 V14" reference="global:allDifferent"/>
<constraint name="C14" arity="4" scope="V2 V5 V8 V15" reference="global:allDifferent"/>
<constraint name="C15" arity="4" scope="V3 V6 V9 V12" reference="global:allDifferent"/>
</constraints>
</instance>
