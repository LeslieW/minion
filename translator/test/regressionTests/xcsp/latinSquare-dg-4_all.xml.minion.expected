MINION 3

#This file has been automatically generated by TAILOR v0.2
#bug-reports: andrea@cs.st-and.ac.uk

# amount of common subexpressions used:0
# amount of inferred common subexpressions used:0
# amount of original variables saved (over direct equality):0
# amount of constraints:16
# amount of auxiliary variables: 0

**VARIABLES**
DISCRETE V0 {0..3}
DISCRETE V1 {0..3}
DISCRETE V2 {0..3}
DISCRETE V3 {0..3}
DISCRETE V4 {0..3}
DISCRETE V5 {0..3}
DISCRETE V6 {0..3}
DISCRETE V7 {0..3}
DISCRETE V8 {0..3}
DISCRETE V9 {0..3}
DISCRETE V10 {0..3}
DISCRETE V11 {0..3}
DISCRETE V12 {0..3}
DISCRETE V13 {0..3}
DISCRETE V14 {0..3}
DISCRETE V15 {0..3}

# aliases

**SEARCH**

PRINT [
[V0],[V1],[V2],[V3],[V4],[V5],[V6],[V7],
[V8],[V9],[V10],[V11],[V12],[V13],[V14],[V15]]

VARORDER [
V0,V1,V2,V3,V4,V5,V6,V7,
V8,V9,V10,V11,V12,V13,V14,V15]


**CONSTRAINTS**

alldiff([ V3, V6, V9, V12])
alldiff([ V2, V5, V8, V15])
alldiff([ V1, V4, V11, V14])
alldiff([ V0, V7, V10, V13])
alldiff([ V0, V5, V10, V15])
alldiff([ V4, V9, V14, V3])
alldiff([ V8, V13, V2, V7])
alldiff([ V12, V1, V6, V11])
alldiff([ V3, V7, V11, V15])
alldiff([ V2, V6, V10, V14])
alldiff([ V1, V5, V9, V13])
alldiff([ V0, V4, V8, V12])
alldiff([ V12, V13, V14, V15])
alldiff([ V8, V9, V10, V11])
alldiff([ V4, V5, V6, V7])
alldiff([ V0, V1, V2, V3])

**EOF**
